src/gpuarray_buffer_cuda.c

```diff
@@ -637,6 +643,7 @@ static int cuda_records(gpudata *a, int flags, CUstream s) {
   if (flags & CUDA_WAIT_WRITE)
     a->ctx->err = cuEventRecord(a->wev, s);
   cuda_exit(a->ctx);
+  a->ls = s;
```

Comment:
I agree that we should call cuda_wait/cuda_record in pair. But what about the cuStreamWaitEvent and cuEventRecord. The current shortcut logic break those pair. I think it is fine, but I would like a confirmation.